config/config.go

```diff
@@ -508,6 +522,49 @@ func (c *ServersetSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return checkOverflow(c.XXX, "serverset_sd_config")
 }
 
+// KubernetesSDConfig is the configuration for Kubernetes service discovery.
+type KubernetesSDConfig struct {
+	Server           string   `yaml:"server"`
+	NodeMetricsPort  int      `yaml:"node_metrics_port,omitempty"`
+	InCluster        bool     `yaml:"in_cluster,omitempty"`
+	BearerTokenFile  string   `yaml:"bearer_token_file,omitempty"`
```

Comment:
All those authentication options are very similar to your recent PR for scrape authentication. I assume this is not specific to Kubernetes but because your organisation has the API secured in the same manner as your `/metrics` endpoints?

This would certainly be interesting for other SD mechanisms too, e.g. Consul or serversets.
How about moving this into its own config block and allow plugging it in into other mechanisms more easily?

If so, it would be easiest to remove it from this PR for now and add this it in a later one.